script/distanceOfErrMsg.pl

```diff
@@ -0,0 +1,128 @@
+use Text::JaroWinkler qw(strcmp95);
+use Params::Validate qw/validate_pos/;
+use List::Util qw/max min/;
+
+# Function to calculate the JaroWinkler distance
+sub JWdistance {
+    my ($s1, $s2) = validate_pos(@_, 1, 1);
+    my @s1 = split //, $s1;
+    my @s2 = split //, $s2;
+
+    my ($max, $min);
+    if (@s1 > @s2) {
+        ($max, $min) = (\@s1, \@s2);
+    } else {
+        ($max, $min) = (\@s2, \@s1);
+    }
+
+    my $max_len = scalar @$max;
+    my $min_len = scalar @$min;
+
+    my $range = max($max_len / 2 - 1, 0);
+
+    my $m_indexes = [];
+    for (my $i = 0; $i < $min_len; $i++) {
+        $m_indexes->[$i] = -1;
+    }
+
+    my $m_flags   = [];
+    for (my $i = 0; $i < $max_len; $i++) {
+        $m_flags->[$i] = undef;
+    }
+
+    my $matches   = 0;
+    for (my $mi = 0; $mi < $min_len; $mi++) {
+        my $c1 = $min->[$mi];
+
+        my $xi = max($mi - $range, 0);
+        my $xn = min($mi + $range + 1, $max_len);
+
+        for (; $xi < $xn; $xi++) {
+            if (!$m_flags->[$xi] && $c1 eq $max->[$xi]) {
+                $m_indexes->[$mi] = $xi;
+                $m_flags->[$xi] = 1;
+                $matches++;
+                last;
+            }
+        }
+    }
+
+    my $ms1 = [];
+    my $ms2 = [];
+
+    for (my ($i, $si) = (0, 0); $i < $min_len; $i++) {
+        if ($m_indexes->[$i] != -1) {
+            $ms1->[$si] = $min->[$i];
+            $si++;
+        }
+    }
+
+    for (my ($i, $si) = (0, 0); $i < $max_len; $i++) {
+        if ($m_flags->[$i]) {
+            $ms2->[$si] = $max->[$i];
+            $si++;
+        }
+    }
+
+    my $transpositions = 0;
+    for (my $mi = 0; $mi < @$ms1; $mi++) {
+        if ($ms1->[$mi] ne $ms2->[$mi]) {
+            $transpositions++;
+        }
+    }
+
+    my $prefix = 0;
+    for (my $mi = 0; $mi < $min_len; $mi++) {
+        if ($s1[$mi] eq $s2[$mi]) {
+            $prefix++;
+        } else {
+            last;
+        }
+    }
+
+    if ($matches == 0) {
+        return 0;
+    }
+    my $t = sprintf "%d", $transpositions/ 2;
+
+    my $j = (($matches / @s1) + ($matches / @s2) + (($matches - $t) / $matches)) / 3;
+    my $jw = $j < 0.7 ? $j : $j + min(0.1, 1 / @$max) * $prefix * (1 - $j);
+    return $jw;
+}
+
+my @existErrMsg = "test"; # Existing error message
+
+my $dir = "$OPENQA_BASEDIR/openqa/testresults/*"; # result dir
+my @files = glob( $dir );
+my @newfiles = undef;
+foreach my $temp (@files) {
+    @newfiles = glob($temp . "/autoinst-log.txt");
+    foreach my $t (@newfiles) {
+        print $t . "\n";
+    }
+}
+
+my $line = undef; # lines in autoinst-log.txt
+my %Lresults = (); # used to store the Levenshtein result
+foreach my $path (@newfiles) {
+    open (FILE, $path);
+    while ($line = <FILE>) {
+        my $where = index($line, "Test died");
+        if ($where != -1) {
+            my $errmsg = substr($line, $where);
+            print $errmsg . "\n";
+            print distance($existErrMsg, $errmsg) . "\n";

```

Comment:
```suggestion
            say $errmsg;
            say distance($existErrMsg, $errmsg);
```
But it would also make sense to write this to stderr so there's a distinction between debug output and actual results.